sdks/python/apache_beam/runners/worker/operations.py

```diff
@@ -496,7 +496,8 @@ def __init__(self, name_context, spec, counter_factory, state_sampler):
     fn, args, kwargs = pickler.loads(self.spec.combine_fn)[:3]
     self.combine_fn = curry_combine_fn(fn, args, kwargs)
     if (getattr(fn.add_input, 'im_func', None)
-        is core.CombineFn.add_input.__func__):
+        is getattr(core.CombineFn.add_input, '__func__',
```

Comment:
I am not sure I follow the logic here. Could we instead say:
```
if (fn.add_input.__func__ is core.CombineFn.add_input.__func__):
   ...
```
cc: @robertwb .

Reply:
On Python 2, getting a method from a class returns an unbound method, with a reference to the wrapped function stored in `__func__`.
On Pyhon 3, getting a method from a class returns the function itself.

On Python 2.6+, `im_func` is also available as `__func__`, so we can change it to
```Python
if (getattr(fn.add_input, '__func__', None)
    is is getattr(core.CombineFn.add_input, '__func__',
                  core.CombineFn.add_input)):
```
if this is more clear.

Reply:
Thanks for the explanation. Upon a closer look at this code, I think the whole `if` section is legacy code and should be removed.

Reply:
In other words, we always follow `else` branch.